### ISSUE TITLE
odsComponentStageScanWithAqua does not use openShift.getImageReference method

### ISSUE THREAD

**Issue by gerardcl** [opendevstack/ods-jenkins-shared-library#918]
**Describe the bug**
In the cases (e.g.: when having multiple components in a repo) that one requires modifying the `resourceName` (hence not matching `componentId`) then `odsComponentStageScanWithAqua` is not able to scan the resulting image because it does not find the reference, and that is due to a different approach when trying to get the `ImageRef`.

**To Reproduce**
1. Provision a quickstarter that builds images on openshift
2. Change the resourceName in the build attributes to not match the component ID
3. Commit and push the code
4. See error logs below

**Expected behavior**
We should align the way we retrieve the `ImageRef` everywhere. There is the 'OpenShiftService' that provides that method `getImageReference ` and it is used in `BuildOpenShiftImageStage ` (https://github.com/opendevstack/ods-jenkins-shared-library/blob/2b8eca74e341e31c3243b588ba9c9efb4ef07a15/src/org/ods/component/BuildOpenShiftImageStage.groovy#L135) or `odsComponentFindOpenShiftImageOrElse` (https://github.com/opendevstack/ods-jenkins-shared-library/blob/2b8eca74e341e31c3243b588ba9c9efb4ef07a15/vars/odsComponentFindOpenShiftImageOrElse.groovy#L39).
Instead, in `ScanWithAquaStage` we rely on the `context.getBuildArtifactURIs().builds` map which does not find `options.resourceName` in the cases like the exposed above (https://github.com/opendevstack/ods-jenkins-shared-library/blob/2b8eca74e341e31c3243b588ba9c9efb4ef07a15/src/org/ods/component/ScanWithAquaStage.groovy#L134).

**Screenshots**
N/A

**Affected version (please complete the following information):**
 - OpenShift: all
 - OpenDevStack > 4.x
 
**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
[Pipeline] { (Aqua Security Scan)
[Pipeline] echo
Skipping as imageRef could not be retrieved. Possible reasons are:
-> The aqua stage runs before the image build stage and hence no new image was created yet.
-> The image build stage was not executed because the image was imported.
-> The aqua stage and the image build stage have different values for 'resourceName' set.
```

**Additional context**
We should change this behaviour unless someone can tell if there is another constraint we need to deal with (after checking the code/tests it looks like there might be some dependencies/issues to be handled while providing docs/reports).
My question is: do we really support mono-repo/multi-component from release-manager point of view? if not then we have nothing to do here and we would need to split onto multiple repos.

FYI @braisvq1996 @hrcornejo

**Comment by gerardcl** [comment]
@metmajer @henrjk FYI

**Comment by metmajer** [comment]
@gerardcl apologies, this slipped through. Can you imagine to provide a suitable implementation? FYI @hrcornejo

**Comment by clemensutschig** [comment]
@serverhorror has a fix for this I thinkVon meinem iPhone gesendetAm 06.01.2023 um 10:12 schrieb gerardcl ***@***.***>:﻿
Hi @metmajer! and happy new year everyone! 🎊
Happy to help here, but would be good to understand the current implementation's rationale (see Additional context in the description) to be sure we are not missing something and to be all aligned.
Currently, we plan activities regarding aqua-cli and trivy comparisons/migrations and we could align there.
Also, FYI we just merged into ods-pipeline's master the first trivy CLI integration focused on generating SBOMs for the generated images (see opendevstack/ods-pipeline#592).

**Comment by gerardcl** [comment]
hi @clemensutschig  and @metmajer!

I have spoken with @serverhorror, and here I have found the key: 
https://github.com/opendevstack/ods-jenkins-shared-library/pull/916/files#diff-96df04d4dcbce1a147ac0981b3c1733485a943bb165298cc4be3e61c768b90cdR44 

So, my suggestions is to wait for that related PR#916 to be merged, and then make the build stage to properly define the target `resourceName` when calling aqua. That should be it (and of course updating the tests that are still pending on the methods described in the description above).

**Comment by gerardcl** [comment]
After testing recent master merge  of PR #916, we can close this issue since https://github.com/opendevstack/ods-jenkins-shared-library/pull/916/files#diff-8ac182f5eaada3eedc19f3865224b0a5c7abf04ca89cc6b2be572844b1e31d77R83 

Fixes it, we have now again support for multiple components to be scanned by aqua implicitly when on build stage.

Thanks a ton @serverhorror !!!

**Comment by gerardcl** [comment]
@braisvq1996  should we add this somehow in the changelog too? so in the next release this fix also gets included although being implicitly fixed by another issue's PR?

**Comment by braisvq1996** [comment]
@gerardcl yeah, we can add a new line in the Changelog to indicate that this issue was fixed
